raw-mm.valid.main.lua

```diff
@@ -3907,43 +3921,43 @@ function defs.got_trueblind()
 end
 
 function defs.got_truedeaf()
-  local result = checkany (dict.truedeaf.herb, dict.attraction.herb, dict.deaf.herb)
+  local result = checkany (dict.truedeaf.herb, dict.attraction.herb, dict.deaf.herb,dict.truedeaf.wafer, dict.deaf.wafer)
 
   if not result and conf.aillusion then
     return
   elseif not result and not conf.aillusion then
-    checkaction (dict.truedeaf.herb, true)
-    lifevision.add(actions.truedeaf_herb.p)
+    checkaction (dict.truedeaf.wafer, true)
+    lifevision.add(actions.truedeaf_wafer.p)
     return
   end
 
   herb_cure = true
   if result.action_name == "truedeaf" then
-    lifevision.add(actions.truedeaf_herb.p)
+    lifevision.add(actions.truedeaf_wafer.p)
 
     if affs.attraction then
-      checkaction (dict.attraction.herb, true)
-      lifevision.add(actions.attraction_herb.p)
+      checkaction (dict.attraction.wafer, true)
+      lifevision.add(actions.attraction_wafer.p)
     end
   else
-    killaction(dict[result.action_name].herb)
-    checkaction (dict.truedeaf.herb, true)
-    checkaction (dict.attraction.herb, true)
-    lifevision.add(actions.truedeaf_herb.p)
-    lifevision.add(actions.attraction_herb.p)
+    killaction(dict[result.action_name].wafer)
+    checkaction (dict.truedeaf.wafer, true)
+    checkaction (dict.attraction.wafer, true)
+    lifevision.add(actions.truedeaf_wafer.p)
+    lifevision.add(actions.attraction_wafer.p)
   end
 end
 
 function defs.cureddeaf()
-  local result = checkany (dict.truedeaf.herb, dict.deaf.herb)
+  local result = checkany (dict.truedeaf.herb, dict.deaf.herb, dict.truedeaf.wafer, dict.deaf.wafer)
```

Comment:
If you'll only support wafer like below, might as well take the .herb equivalents out from here.